aesm-client/tests/live_quote.rs

```diff
@@ -41,3 +44,61 @@ fn live_quote() {
         )
         .expect("quote result");
 }
+
+pub fn get_algorithm_id(key_id : &Vec<u8>) -> u32 {
+    const ALGORITHM_OFFSET : usize = 154;
+
+    let mut bytes: [u8; 4] = Default::default();
+    bytes.copy_from_slice(&key_id[ALGORITHM_OFFSET..ALGORITHM_OFFSET+4]);
+    u32::from_le_bytes(bytes)
+}
+
+#[cfg(not(windows))]
+#[test]
+fn test_ecdsa_quote() {
+    const NONCE: [u8; 16] = [0; 16];
+    const SGX_QL_ALG_ECDSA_P256 : u32 = 2;
+
+    let mut device = IsgxDevice::new()
+        .unwrap()
+        .einittoken_provider(AesmClient::new())
+        .build();
+
+    let client = AesmClient::new();
+
+    let key_ids = client.get_supported_att_key_ids().unwrap();
+
+    // Print the keys for debugging
+    for i in key_ids.iter() {
+        let ptr = i.as_ptr() as *const sgx_att_key_id_ext_t;
```

Comment:
If you only use `sgx_att_key_id_ext_t` for debugging, then you might as well not introduce the dependency

Reply:
removed it entirely